pkg/cluster/fetch.go

```diff
@@ -23,20 +23,27 @@ type EksdReleaseFetch func(ctx context.Context, name, namespace string) (*eksdv1
 
 type OIDCFetch func(ctx context.Context, name, namespace string) (*v1alpha1.OIDCConfig, error)
 
-func BuildSpecForCluster(ctx context.Context, cluster *v1alpha1.Cluster, bundlesFetch BundlesFetch, eksdReleaseFetch EksdReleaseFetch, gitOpsFetch GitOpsFetch, oidcFetch OIDCFetch) (*Spec, error) {
-	bundles, err := GetBundlesForCluster(ctx, cluster, bundlesFetch)
+type Fetcher interface {
```

Comment:
Why force an implementation to know about all these things? When we want another 'thing' we have to extend the interface which is bad. Perhaps you're looking for a config object?

```go
type BuildSpecForClusterConfig struct {
  BundlesFetcher BundlesFetcher
  ...
}

func BuildSpecForClsuter(context.Context, BuildSpecForClusterConfig) ...
```